src/test/regress/sql/citus_local_tables_mx.sql

```diff
@@ -464,6 +474,63 @@ select run_command_on_workers($$SELECT count(*) from citus_local_tables_mx.v100$
 	select run_command_on_workers($$SELECT count(*) from citus_local_tables_mx.v101$$);
 	select run_command_on_workers($$SELECT count(*) from citus_local_tables_mx.v102$$);
 
+-- test a matview with columnar
+CREATE MATERIALIZED VIEW matview_columnar USING COLUMNAR AS SELECT * FROM loc_tb;
+
+-- test REFRESH MAT VIEW
+INSERT INTO loc_tb VALUES (1), (2);
+SELECT citus_add_local_table_to_metadata('loc_tb', true);
+SELECT * FROM matview_101 ORDER BY a;
+REFRESH MATERIALIZED VIEW matview_101;
+SELECT * FROM matview_101 ORDER BY a;
+
+-- verify columnar matview works on a table added to metadata
+SELECT * FROM matview_columnar;
+REFRESH MATERIALIZED VIEW matview_columnar;
+SELECT * FROM matview_columnar ORDER BY a;
+
+-- test with partitioned tables
+SET citus.use_citus_managed_tables TO ON;
+CREATE TABLE parent_1 (a INT UNIQUE) PARTITION BY RANGE(a);
+SET citus.use_citus_managed_tables TO OFF;
+
+CREATE MATERIALIZED VIEW part_matview1 as SELECT count(*) FROM parent_1 JOIN parent_1 p2 ON (true);
+CREATE MATERIALIZED VIEW part_matview2 as SELECT count(*) FROM parent_1 JOIN part_matview1 on (true);
+
+SELECT count(*) FROM citus_local_tables_mx.part_matview1 JOIN citus_local_tables_mx.part_matview2 ON (true);
+
+CREATE TABLE parent_1_child_1 (a int);
+CREATE TABLE parent_1_child_2 (a int);
+
+-- create matviews on partition tables
+CREATE MATERIALIZED VIEW mv1 AS SELECT * FROM parent_1_child_1;
+CREATE MATERIALIZED VIEW mv2 AS SELECT * FROM parent_1_child_2;
+CREATE MATERIALIZED VIEW mv3 AS SELECT parent_1_child_2.* FROM parent_1_child_2 JOIN parent_1_child_1 USING(a);
+CREATE MATERIALIZED VIEW mv4 AS SELECT * FROM mv3;
+
+alter table parent_1 attach partition parent_1_child_1 FOR VALUES FROM (0) TO (10) ;
+
+-- all matviews work
+SELECT count(*) FROM citus_local_tables_mx.mv1;
+SELECT count(*) FROM citus_local_tables_mx.mv2;
+SELECT count(*) FROM citus_local_tables_mx.mv3;
+SELECT count(*) FROM citus_local_tables_mx.mv4;
+
+-- recreate matviews and verify they still work
+alter table parent_1 attach partition parent_1_child_2 FOR VALUES FROM (10) TO (20);
+
+SELECT count(*) FROM citus_local_tables_mx.mv1;
+SELECT count(*) FROM citus_local_tables_mx.mv2;
+SELECT count(*) FROM citus_local_tables_mx.mv3;
+SELECT count(*) FROM citus_local_tables_mx.mv4;
+
+-- verify matviews work after undistributing
+SELECT undistribute_table('parent_1');
+SELECT count(*) FROM citus_local_tables_mx.mv1;
+SELECT count(*) FROM citus_local_tables_mx.mv2;
+SELECT count(*) FROM citus_local_tables_mx.mv3;
+SELECT count(*) FROM citus_local_tables_mx.mv4;
+
```

Comment:
Let's add a comment about unsupported CREATE MATERIALIZED VIEW options to parse https://github.com/citusdata/citus/issues/5968. So we can add more tests here once that issue is fixed.